### PR TITLE
Keb split provisioning initialisation

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- use new provisioning manager (StagedManager) - provide 4 stages
- split provisioning initialisation step into smaller parts
- reduce InstanceStorage usage by putting DashboardURL to the operation
- change the staged maanger to not put the operation to the queue after the step returns a retry

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/provisioning/ias_type_step.go [hunk 1]
```diff
@@ -14,62 +15,69 @@ const (
...
-		err = spb.ConfigureServiceProviderType(runtimeURL)
+		log.Infof("Configure SSO Type for ServiceProvider %q with RuntimeURL: %s", spb.ServiceProviderName(), operation.DashboardURL)
+		err = spb.ConfigureServiceProviderType(operation.DashboardURL)
 		if err != nil {
 			return s.handleError(operation, err, log, "setting SSO Type failed")
 		}
 	}
 
-	return 0, nil
+	grafanaURL := fmt.Sprintf("https://grafana.%s", operation.ShootDomain)
```
Comment (jasiu001) [inline components/kyma-environment-broker/internal/process/provisioning/ias_type_step.go:56]: it has nothing to do with IAS

File: components/kyma-environment-broker/internal/process/provisioning/start_step.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			if err != nil {
+				if dberr.IsNotFound(err) {
+					log.Errorf("Instance does not exists.")
+					return s.operationManager.OperationFailed(operation, "The instance does not exists", log)
+				}
+				log.Errorf("Unable to get the instance: %s", err.Error())
+				return operation, time.Second, nil
+			}
+			log.Infof("Setting the newest InstanceDetails")
+			operation.InstanceDetails = inst.InstanceDetails
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/start_step.go:58]: use GetInstanceDEtails method!!!

File: components/kyma-environment-broker/internal/process/provisioning/check_runtime_step.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	case gqlschema.OperationStatePending:
+		return operation, 2 * time.Minute, nil
+	case gqlschema.OperationStateFailed:
+		return s.operationManager.OperationFailed(operation, fmt.Sprintf("provisioner client returns failed status: %s", msg), log)
+	}
+
+	return s.operationManager.OperationFailed(operation, fmt.Sprintf("unsupported provisioner client status: %s", status.State.String()), log)
+}
+
+func (s *CheckRuntimeStep) handleDashboardURL(operation *internal.ProvisioningOperation, log logrus.FieldLogger) (time.Duration, error) {
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/process/provisioning/check_runtime_step.go:91]: I've just take it from initialisation step, but you are right, we can have another step. the idea of checking the URL is that the URL in the director is set by Kyma and it means Kyma is installed

File: components/kyma-environment-broker/internal/process/provisioning/create_runtime.go [hunk 4]
```diff
@@ -46,6 +46,9 @@ func (s *CreateRuntimeStep) Name() string {
 }
 
 func (s *CreateRuntimeStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if operation.RuntimeID != "" {
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/process/provisioning/create_runtime.go:49]: I'll redesign the step

File: components/kyma-environment-broker/internal/process/provisioning/ias_type_step.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/process/provisioning/ias_type_step.go:56]: moved to the CreateRuntime step

File: components/kyma-environment-broker/internal/process/provisioning/check_runtime_step.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/process/provisioning/check_runtime_step.go:91]: new step is created

**Review by jasiu001 — DISMISSED** [review]

**Review by jasiu001 — APPROVED** [review]

**Review by jasiu001 — APPROVED** [review]

File: docs/kyma-environment-broker/03-03-runtime-operations.md [hunk 5]
```diff
@@ -15,18 +15,24 @@ Kyma Environment Broker allows you to configure operations that you can run on a
...
+| check_runtime  | Check_Runtime                          | Provisioning             | Checks the status of Provisioner process and asks the Director for the Dashboard URL if the provisioning in Gardener is finished. |  @piotrmiskiewicz (Team Gopher) | 
+| post_actions   | AVS_Create_External_Eval_Step          | AvS                      | Sets up external monitoring of Kyma Runtime.                                                                                      | @piotrmiskiewicz (Team Gopher) |
+| post_actions   | AVS_Tags                               | AvS                      | Sets up proper tags in the internal monitoring system.                                                                            | @piotrmiskiewicz (Team Gopher) |
+| post_actions   | IAS_Type                               | Identity Authentication Service | Configures OIDC authentication.                                                                                            | @piotrmiskiewicz (Team Gopher) |
```
Comment (kazydek) [inline docs/kyma-environment-broker/03-03-runtime-operations.md:33]:
```suggestion
| post_actions   | IAS_Type                               | Identity Authentication Service | Configures OpenID Connect (OIDC) authentication.                                                                                            | @piotrmiskiewicz (Team Gopher) |
```

Comment (piotrmiskiewicz) [reply docs/kyma-environment-broker/03-03-runtime-operations.md:33]: ok for me

**Review by kazydek — APPROVED** [review]

**Review by jasiu001 — APPROVED** [review]

**Pull request merged**